validator/src/journal/block_manager.rs

```diff
@@ -66,6 +66,24 @@ impl BlockRef {
     pub fn block_id(&self) -> &str {
         &self.block_id
     }
+
+    pub fn get_block(&self) -> Block {
```

Comment:
BlockRef should be renamed if it's a pass-through interface to the BlockManager.

Before this change, BlockRef's primary duty was to allow holding a reference to a chain head. After this change, it's not clear what the primary job of BlockRef, as it is both pass-through and ref counting. Seems like a design regression in it's added conceptual complexity.

This change is also very tied to our current BlockRef implementation that directly holds a BlockManager. It is not clear that BlockRef should even have access to anything other than unref_block wrt BlockManager (if we exclude this new get_block function).

Reply:
Makes sense to me. I removed this commit and modified the chain controller to just use its own `get_block` method